mailpoet/lib/Subscription/Registration.php

```diff
@@ -56,9 +56,22 @@ public function extendForm() {
       </label>
     </p>';
 
-    $form = $this->wp->applyFilters('mailpoet_register_form_extend', $form);
-
-    print $form;
+    $form = (string)$this->wp->applyFilters('mailpoet_register_form_extend', $form);
+    $allowedHtml = [
+      'p' => [
+        'class' => true,
+      ],
+      'label' => [
+        'for' => true,
+      ],
+      'input' => [
+        'type' => true,
+        'id' => true,
+        'value' => true,
+        'name' => true,
+      ],
+    ];
+    print wp_kses($form, $allowedHtml);
```

Comment:
`wp_kses_post()` does not allow for `input`, this is why i switched to `wp_kses()`. We could use the wp_kses_post array together with input though :thinking:

Reply:
I guess we should add an ignore rule here. Our HTML is safe and any modification by the hooks will probably need to be correctly sanitized by the user of the hook as we can't predict what 3rd parties may want to use.

Reply:
Agree.

Reply:
> We could use the wp_kses_post array together with input though
Do you mean combining output from `wp_kses_allowed_html('post')` with our custom array?

Reply:
Yes. But this would for example not allow for `<script>` but it is not impossible that a custom integration needs `<script>`. This is why I would agree with @JanJakes and conclude: Our template is good. If a 3rd party decided to inject malicious stuff through this filter: They do not need this filter to do so, they can execute PHP so everything is lost already.